ios/RCCNavigationController.m

```diff
@@ -272,10 +274,27 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
     NSString *title = button[@"title"];
     UIImage *iconImage = nil;
     id icon = button[@"icon"];
+    NSString *customView = button[@"customView"];
     if (icon) iconImage = [RCTConvert UIImage:icon];
-    
+    NSString *buttonId = button[@"id"];
+    NSString *callbackId = button[@"onPress"];
+
     UIBarButtonItem *barButtonItem;
-    if (iconImage)
+    if (customView) {
+      RCTBridge *bridge = ((RCTRootView*)viewController.view).bridge;
+
+      NSNumber *viewWidth = button[@"customViewWidth"] ?: @60;
```

Comment:
I'm pretty 👎 on giving a specific view size here, rather than self sizing. I think I'd prefer to see custom views have their own width and height as derived by RN/Yoga.